src/fields.jl

```diff
@@ -52,3 +60,29 @@ Field(LX, data, grid::Grid1D) = Field1D(LX, data, grid)
 Constructs a 2D field of `data` at location `(LX, LY)` on `grid`.
 """
 Field(LX, LY, data, grid::Grid2D) = Field2D(LX, LY, data, grid)
+
+function fill_halos!(input::Field1D{Centre, Grid1D{Periodic}}, data::AbstractArray)
```

Comment:
In my head `fill_halos!` **does not require** any data input. See my propose here https://github.com/ClimateFluidPhysics-ANU/MixedLayerThermoclineDynamics.jl/issues/8#issuecomment-882014866

Why is `data` needed to fill the halos? Let's think of what "filling halos" involve and if that requires `data` then fair enough. But I'm not convinced it does.

Reply:
On second look on the code I see that you are using `fill_halos!()` to fill up the non-halo values of the field.
Why do we need to do that? --- It's misleading! Imagine if the function `interpolation!` does derivatives and `∂x!` does interpolation...

Reply:
May I suggest something in the spirit of:

```Julia
function Field1D(LX, data::AbstractArray, grid::Grid1D)
    data_with_halos = OffsetArray(zeros(grid.nx + 2*grid.hx), -grid.hx)
    @. data_with_halos[1:nx] = data

    field = Field1D{LX, typeof(grid)}(data_with_halos, grid)

    fill_halos!(field)

    return field
end

function fill_halos!(field::Field1D{Face, Grid1D{Periodic}})
     nx, hx, dx = field.grid.nx, field.grid.hx, field.grid.dx

     for j in 1:hx
        field.data[nx+j] = field.data[j]
        field.data[-j+1] = field.data[nx-j+1]
    end

    return nothing
 end
```

Some remarks:
- This way `fill_halos!` only fills the halos so its name is descriptive.
- I renamed `new_data` to `data_with_halos` in an attempt to be descriptive. `new_data` reveals as little information as possible while with 6 more character one convey the intention of this new variable which becomes helpful when somebody is trying to understand what the code is doing in an attempt to improve/fix it.
- In the same spirit as above, I renamed the argument of `fill_halos!` from input to `field`.